src/main/java/org/mule/extension/file/common/api/source/AbstractPostActionGroup.java

```diff
@@ -59,24 +55,33 @@ public void apply(AbstractFileSystem fileSystem, FileAttributes fileAttributes,
       }
     }
 
-    boolean moved = false;
+    boolean movedOrRenamed = false;
     try {
       if (getMoveToDirectory() != null) {
         fileSystem.move(config, fileAttributes.getPath(), getMoveToDirectory(), false, true,
                         getRenameTo());
-        moved = true;
+        movedOrRenamed = true;
+      } else if (getRenameTo() != null) {
+        fileSystem.rename(fileAttributes.getPath(), getRenameTo(), false);
+        movedOrRenamed = true;
       }
     } catch (FileAlreadyExistsException e) {
       if (!isAutoDelete()) {
-        String moveToFileName = getRenameTo() == null ? fileAttributes.getName() : getRenameTo();
-        String moveToPath = Paths.get(getMoveToDirectory()).resolve(moveToFileName).toString();
-        LOGGER.warn(String.format("A file with the same name was found when trying to move '%s' to '%s'" +
-            ". The file '%s' was not sent to the moveTo directory and it remains on the poll directory.",
-                                  fileAttributes.getPath(), moveToPath, fileAttributes.getPath()));
+        if (getMoveToDirectory() == null) {
+          LOGGER.warn(String.format("A file with the same name was found when trying to rename '%s' to '%s'" +
+              ". The file '%s' was not renamed and it remains on the poll directory.",
+                                    fileAttributes.getName(), getRenameTo(), fileAttributes.getPath()));
+        } else {
+          String moveToFileName = getRenameTo() == null ? fileAttributes.getName() : getRenameTo();
+          String moveToPath = Paths.get(getMoveToDirectory()).resolve(moveToFileName).toString();
+          LOGGER.warn(String.format("A file with the same name was found when trying to move '%s' to '%s'" +
```

Comment:
We try to avoid using Class.method and use only the method and we add a static import.
Also in this case you can use the following:
`LOGGER.warn("A file with the same name was found when trying to move {} to {}. The file {} was not sent to the moveTo directory and it remains on the poll directory.", fileAttributes.getPath(), moveToPath, fileAttributes.getPath()))`